lnwallet/channel.go

```diff
@@ -2466,6 +2474,10 @@ func (lc *LightningChannel) fetchCommitmentView(remoteChain bool,
 	return c, nil
 }
 
+func (lc *LightningChannel) fundingTxIn() wire.TxIn {
```

Comment:
Why add this overhead of creating a _new_ funding txin each time? I dig the rest of the refactoring here, but this seems unnecessary.